KIPs/kip-245.md

```diff
@@ -163,23 +160,20 @@ for tx in bundle:
   ApplyTransaction(tx)
   if tx failed:
     RevertToSnapshot(preBundleSnapshot)
-  else if tx is the last in bundle:
-    append all txs and receipts in bundle to env.txs and env.receipts
+    return
+append all txs and receipts in bundle to env.txs and env.receipts
 ```
 
-### Delaying Timeout
+#### Pop transactions
 
-Timeout handler must be delayed if a bundle is currently being executed.
+Currently, if transaction execution fails, there are cases when all dependent transactions are removed (from `ApplyTransactions` loop) as they cannot be executed in this block.
+Such cases include gas limit reached, too high nonce, unsupported transaction type, etc.
+Likewise, if a transaction in a bundle fails to execute, all dependent transactions must be removed.
+Transactions are dependent if they have the same sender, or they are in the same bundle.
```

Comment:
How about providing a concrete example to help understanding?
```
txs =     [ G1 A.T1 B.T1 A.T2 B.T2 C.T1 ]
bundle =  [ G1 A.T1 ]
bundle2 = [ B.T1 A.T2 ]
bundle3 = [ B.T2 C.T1 ]
on A.T1 revert, pop(txs, 2) => txs = [ ]
```

Reply:
added b0d3742